packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -4,322 +4,70 @@
-          new EventListener() {
-
-            @Override
-            public void onPlayerStateChanged(final boolean playWhenReady, final int playbackState) {
-              if (playbackState == Player.STATE_BUFFERING) {
-                sendBufferingUpdate();
-              } else if (playbackState == Player.STATE_READY) {
-                if (!isInitialized) {
-                  isInitialized = true;
-                  sendInitialized();
-                }
-              } else if (playbackState == Player.STATE_ENDED) {
-                Map<String, Object> event = new HashMap<>();
-                event.put("event", "completed");
-                eventSink.success(event);
-              }
-            }
-
-            @Override
-            public void onPlayerError(final ExoPlaybackException error) {
-              if (eventSink != null) {
-                eventSink.error("VideoError", "Video player had error " + error, null);
-              }
-            }
-          });
-
-      Map<String, Object> reply = new HashMap<>();
-      reply.put("textureId", textureEntry.id());
-      result.success(reply);
-    }
 
-    private void sendBufferingUpdate() {
-      Map<String, Object> event = new HashMap<>();
-      event.put("event", "bufferingUpdate");
-      List<? extends Number> range = Arrays.asList(0, exoPlayer.getBufferedPosition());
-      // iOS supports a list of buffered ranges, so here is a list with a single range.
-      event.put("values", Collections.singletonList(range));
-      eventSink.success(event);
-    }
-
-    @SuppressWarnings("deprecation")
-    private static void setAudioAttributes(SimpleExoPlayer exoPlayer) {
-      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-        exoPlayer.setAudioAttributes(
-            new AudioAttributes.Builder().setContentType(C.CONTENT_TYPE_MOVIE).build());
-      } else {
-        exoPlayer.setAudioStreamType(C.STREAM_TYPE_MUSIC);
-      }
-    }
-
-    void play() {
-      exoPlayer.setPlayWhenReady(true);
-    }
-
-    void pause() {
-      exoPlayer.setPlayWhenReady(false);
-    }
-
-    void setLooping(boolean value) {
-      exoPlayer.setRepeatMode(value ? REPEAT_MODE_ALL : REPEAT_MODE_OFF);
-    }
+/** Android platform implementation of the VideoPlayerPlugin. */
+public class VideoPlayerPlugin implements MethodCallHandler, FlutterPlugin {
+  private static final String TAG = "VideoPlayerPlugin";
+  private final LongSparseArray<VideoPlayer> videoPlayers = new LongSparseArray<>();
+  private FlutterState flutterState;
 
-    void setVolume(double value) {
-      float bracketedValue = (float) Math.max(0.0, Math.min(1.0, value));
-      exoPlayer.setVolume(bracketedValue);
-    }
-
-    void seekTo(int location) {
-      exoPlayer.seekTo(location);
-    }
-
-    long getPosition() {
-      return exoPlayer.getCurrentPosition();
-    }
+  /** Register this with the v2 embedding for the plugin to respond to lifecycle callbacks. */
+  public VideoPlayerPlugin() {}
```

Comment:
Seems not necessary

Reply:
Yeah, it's redundant. I think I'm fine having this just as a place to hang the documentation on though.